core/trino-main/src/main/java/io/trino/execution/scheduler/EventDrivenFaultTolerantQueryScheduler.java

```diff
@@ -1190,6 +1195,22 @@ private void createStageExecution(SubPlan subPlan, boolean rootFragment, Map<Sta
             }
         }
 
+        private boolean isNoMemoryFragment(PlanFragment fragment)
+        {
+            if (!fragment.getRemoteSourceNodes().stream().flatMap(node -> node.getSourceFragmentIds().stream())
+                    .allMatch(sourceFragmentId -> stageExecutions.get(getStageId(sourceFragmentId)).isNoMemoryFragment())) {
+                return false;
```

Comment:
why remote sources matter?
shouldn't `isNoMemoryFragment(PlanFragment)` be a local assessment about the fragment?

is this is about whole execution subtree, let's rename the method to indicate that

Reply:
It is about local fragment. It just uses this bit of information for source stages as a shortcut for "source will produce negligible amount of data". Which is a simplification - but not that much given logic we use to set this bit. I will add a comment.